### PR TITLE
Support hide_progress for top-langs feature

### PR DESCRIPTION
**Pull request by amirhakimnejad**

### CODE REVIEW

**Comment by amirhakimnejad** [comment]
[Working example](https://github.com/amirhakimnejad/amirhakimnejad)

File: api/top-langs.js [hunk 1]
```diff
@@ -77,6 +78,7 @@ export default async (req, res) => {
         border_color,
         locale: locale ? locale.toLowerCase() : null,
         disable_animations: parseBoolean(disable_animations),
+        hide_progress: parseBoolean(hide_progress) ? true : false,
```
Comment (Zo-Bro-23) [inline api/top-langs.js:81]: Why is this needed? Simply `hide_progress: parseBoolean(hide_progress)` should work.

File: docs/readme_cn.md [hunk 2]
```diff
@@ -165,12 +165,17 @@ dark, radical, merko, gruvbox, tokyonight, onedark, cobalt, synthwave, highcontr
...
-- `layout` - 在两个可用布局 `default` & `compact` 间切换
-- `card_width` - 手动设置卡片的宽度 _(number)_
+- `hide_title` - _(boolean)_。 默认值：`false`。
+- `layout` - 在两个可用布局`default` 和 `compact` 之间切换。 默认值：`default`。
+- `card_width` - 手动设置卡片的宽度 _(number)_。 默认 `300`。
+- `langs_count` - 在卡片上显示更多语言，在 1-10 _(number)_ 之间。 默认 `5`。
+- `exclude_repo` - 排除指定的存储库_（逗号分隔值）_。 默认值：`[]（空白数组）`。
+- `custom_title` - 为卡片设置自定义标题 _(string)_。 默认“最常用的语言”。
+- `disable_animations` - 禁用卡片中的所有动画_（布尔值）_。 默认值：`false`。
+- `hide_progress` - 它使用紧凑布局选项，隐藏百分比，并使条形相等。
```
Comment (Zo-Bro-23) [inline docs/readme_cn.md:178]: Not needed since we are automating translations now.

File: docs/readme_de.md [hunk 3]
```diff
@@ -157,10 +157,15 @@ Du kannst mehrere, mit Kommas separierte, Werte in der bg_color Option angeben,
...
-- `card_width` - Lege die Breite der Karte manuell fest _(Zahl)_
+- `hide` - Ausblenden der angegebenen Sprachen auf der Karte _(Comma-separated values)_. Standard: `[] (leeres Array)`.
+- `hide_title` - _(boolesch)_. Voreinstellung: „false“.
+- `layout` - Umschalten zwischen zwei verfügbaren Layouts `default` & `compact`. Standard: "default".
+- `card_width` - Stellen Sie die Breite der Karte manuell ein _(number)_. Standard '300'.
+- `langs_count` - Zeige mehr Sprachen auf der Karte, zwischen 1-10 _(number)_. Standard `5`.
+- `exclude_repo` - Angegebene Repositories ausschließen _(Comma-separated values)_. Standard: `[] (leeres Array)`.
+- `custom_title` - Legt einen benutzerdefinierten Titel für die Karte fest _(string)_. Standard `Am häufigsten verwendete Sprachen`.
+- `disable_animations` - Deaktiviert alle Animationen in der Karte _(boolean)_. Voreinstellung: „false“.
+- `hide_progress` - Es verwendet die kompakte Layoutoption, blendet Prozentsätze aus und gleicht die Balken an. Standard `false`.
```
Comment (Zo-Bro-23) [inline docs/readme_de.md:168]: Again, same thing. Please only update the English readme.

File: src/cards/top-languages-card.js [hunk 4]
```diff
@@ -160,18 +163,18 @@ const renderNormalLayout = (langs, width, totalLanguageSize) => {
  * @param {Lang[]} langs Array of programming languages.
  * @param {number} width Card width.
  * @param {number} totalLanguageSize Total size of all languages.
+ * @param {boolean} hideProgress Whether to hide progress bar.
  * @returns {string} Compact layout card SVG object.
  */
-const renderCompactLayout = (langs, width, totalLanguageSize) => {
+const renderCompactLayout = (langs, width, totalLanguageSize, hideProgress) => {
```
Comment (Zo-Bro-23) [inline src/cards/top-languages-card.js:169]: You're using `hideProgress` in some places and `hide_progress` in other places. The GitHub mobile app sucks, so I'm not really sure if you let one equal the other somewhere, but either way, it would be better to have only `hide_progress` everywhere. We use `camelCase` for internal variables and `snake_case` for query parameters, but internal variables that represent query parameters also use snake case to avoid confusion. Please update your code to do the same!

**Review by Zo-Bro-23 — CHANGES_REQUESTED** [review]
Hi, thanks for the high quality PR! A few things before we merge:

- No need to make changes to the docs in all languages! It's really amazing that you've taken the time to do that, but we're unfortunately in the process of migrating to an automated translation system, so we're not accepting any PRs regarding translated docs. I appreciate that you've taken the time to do that, and I apologize for the miscommunication.
- Please see my comment above regarding variable names.
- I would think this is a valuable PR, but I'm not sure about the use case. Why would someone want to make the progress bars equal? It would be misleading to think that they spent equal time on all their languages. Hiding the percentages is a good feature, but again, I'm not sure about switching to compact mode. If the equal bars feature is removed, hiding percentages would make as much sense in the default layout as in the compact layout. Please consider my suggestion! @rickstaa please feel free to add your thoughts...

File: src/cards/top-languages-card.js (hunk 4 above)
Comment (Zo-Bro-23) [reply src/cards/top-languages-card.js:169]: Okay, I checked the code out on my computer, and it seems fine!

File: api/top-langs.js (hunk 1 above)
Comment (amirhakimnejad) [reply api/top-langs.js:81]: Yes, it's true. I just wanted to prevent undefined values. And somehow explicitly say the default value for it.

File: docs/readme_de.md (hunk 3 above)
Comment (amirhakimnejad) [reply docs/readme_de.md:168]: > Hi, thanks for the high quality PR! A few things before we merge:
Hello, and thank you.

> * No need to make changes to the docs in all languages! It's really amazing that you've taken the time to do that, but we're unfortunately in the process of migrating to an automated translation system, so we're not accepting any PRs regarding translated docs. I appreciate that you've taken the time to do that, and I apologize for the miscommunication.
Oh, okay, the translation system seems incredible.

> * I would think this is a valuable PR, but I'm not sure about the use case. Why would someone want to make the progress bars equal? It would be misleading to think that they spent equal time on all their languages. Hiding the percentages is a good feature, but again, I'm not sure about switching to compact mode. If the equal bars feature is removed, hiding percentages would make as much sense in the default layout as in the compact layout. Please consider my suggestion! @rickstaa please feel free to add your thoughts...

About the progress bar. In the `hire_progress=true` scenario, I only thought of it as juice, having the language colors instead of removing the bars. You're right; I can hide it.

About the default mode, I thought they would be so much alike, also, having a language per line without a progress bar looked a bit unnecessarily space-consuming.

I agree with anything you guys decide and will fix the issues.

File: src/cards/top-languages-card.js (hunk 4 above)
Comment (amirhakimnejad) [reply src/cards/top-languages-card.js:169]: > Hi, thanks for the high quality PR! A few things before we merge:

Hello, and thank you.

> * No need to make changes to the docs in all languages! It's really amazing that you've taken the time to do that, but we're unfortunately in the process of migrating to an automated translation system, so we're not accepting any PRs regarding translated docs. I appreciate that you've taken the time to do that, and I apologize for the miscommunication.

Oh, okay, the translation system seems incredible.

> * I would think this is a valuable PR, but I'm not sure about the use case. Why would someone want to make the progress bars equal? It would be misleading to think that they spent equal time on all their languages. Hiding the percentages is a good feature, but again, I'm not sure about switching to compact mode. If the equal bars feature is removed, hiding percentages would make as much sense in the default layout as in the compact layout. Please consider my suggestion! @rickstaa please feel free to add your thoughts...

About the progress bar. In the `hire_progress=true` scenario, I only thought of it as juice, having the language colors instead of removing the bars. You're right; I can hide it.

About the default mode, I thought they would be so much alike, also, having a language per line without a progress bar looked a bit unnecessarily space-consuming.

I agree with anything you guys decide and will fix the issues.

File: api/top-langs.js (hunk 1 above)
Comment (Zo-Bro-23) [reply api/top-langs.js:81]: We already take care of this:
https://github.com/anuraghazra/github-readme-stats/blob/1df150127b6e130195cf0145ebb432a2307e6255/src/common/utils.js#L83-L94

Please remove this line to keep the code nice and clean (it's not needed since we're already accounting for undefined values). Thank you!

Comment (amirhakimnejad) [reply api/top-langs.js]: Done. Thank you

**Comment by amirhakimnejad** [comment]
> Hi, thanks for the high quality PR! A few things before we merge:

Hello, and thank you.

> * No need to make changes to the docs in all languages! It's really amazing that you've taken the time to do that, but we're unfortunately in the process of migrating to an automated translation system, so we're not accepting any PRs regarding translated docs. I appreciate that you've taken the time to do that, and I apologize for the miscommunication.

Oh, okay, the translation system seems incredible.

> * I would think this is a valuable PR, but I'm not sure about the use case. Why would someone want to make the progress bars equal? It would be misleading to think that they spent equal time on all their languages. Hiding the percentages is a good feature, but again, I'm not sure about switching to compact mode. If the equal bars feature is removed, hiding percentages would make as much sense in the default layout as in the compact layout. Please consider my suggestion! @rickstaa please feel free to add your thoughts...

About the progress bar. In the `hire_progress=true` scenario, I only thought of it as juice, having the language colors instead of removing the bars. You're right; I can hide it.

About the default mode, I thought they would be so much alike, also, having a language per line without a progress bar looked a bit unnecessarily space-consuming.

I agree with anything you guys decide and will fix the issues.

Thank you for the useful repo

**Comment by Zo-Bro-23** [comment]
I still think the progress bars should show accurate progress. Otherwise, it will be misleading. I agree that the compact layout makes more sense, but the problem is that this parameter will then essentially _modify_ another parameter, which I don't think should be the case. I still think you should show accurate progress and remove the compact layout check, but @rickstaa and @anuraghazra can override me if they think otherwise.

Also, due to the popularity of the repo, we require at least two reviews before merging, which is why I requested a review from @rickstaa.

**Comment by amirhakimnejad** [comment]
> I still think the progress bars should show accurate progress. Otherwise, it will be misleading. I agree that the compact layout makes more sense, but the problem is that this parameter will then essentially _modify_ another parameter, which I don't think should be the case. I still think you should show accurate progress and remove the compact layout check, but @rickstaa and @anuraghazra can override me if they think otherwise.

Hmmm, maybe hiding the bars would be better. In my own use case, one of my used languages was gaining too much unwanted weight in the parameters, and I wanted to avoid it. But about the accuracy, you're right.
If there is a bar, it should be accurate; otherwise, it should be hidden.

So based on my own case, do you think I should keep the bars accurate with hidden percentages, or hide the bars, too?
Both seem reasonable to me, though. I agree anyways

**Comment by Zo-Bro-23** [comment]
> > I still think the progress bars should show accurate progress. Otherwise, it will be misleading. I agree that the compact layout makes more sense, but the problem is that this parameter will then essentially _modify_ another parameter, which I don't think should be the case. I still think you should show accurate progress and remove the compact layout check, but @rickstaa and @anuraghazra can override me if they think otherwise.
> 
> Hmmm, maybe hiding the bars would be better. In my own use case, one of my used languages was gaining too much unwanted weight in the parameters, and I wanted to avoid it. But about the accuracy, you're right. If there is a bar, it should be accurate; otherwise, it should be hidden.
> 
> So based on my own case, do you think I should keep the bars accurate with hidden percentages, or hide the bars, too? Both seem reasonable to me, though. I agree anyways

Both are fine, but hiding the bars entirely would make more sense, since hiding the labels alone wouldn't do much (you can still guess the percentage based on the bars). Let me play around with the code a bit and see what I can do (I'll make commits directly to your branch).

**Comment by amirhakimnejad** [comment]
> Both are fine, but hiding the bars entirely would make more sense, since hiding the labels alone wouldn't do much (you can still guess the percentage based on the bars). Let me play around with the code a bit and see what I can do (I'll make commits directly to your branch).

Mhm, ok thank you. Let me know if I should change anything.

**Review by Zo-Bro-23 — CHANGES_REQUESTED** [review]
@amirhakimnejad I've made the necessary changes. Please revert the translated docs so that we can merge!

**Comment by amirhakimnejad** [comment]
Thank you, @Zo-Bro-23 , I just pushed the revert.

**Review by Zo-Bro-23 — APPROVED** [review]
We are go for merge 👍

**Review by rickstaa — APPROVED** [review]

**Comment by Zo-Bro-23** [comment]
Merged 🎉
